### ISSUE TITLE
Problems when using images of size 256x256

### ISSUE THREAD

**Issue by claudiom4sir** [sjmoran/CURL#19]
Hi,

I would like to train/test your method on my MIT-Adobe-5K version, which has images of size 256x256 (i.e. the original images have been resized to 256x256). However, I've found some problems related to the value of loss, PSNR and SSIM when computed on the validation/test sets. In particular, it seems that on the train set the loss function decreases, and this is ok, but at validation/test time the value of loss is much higher while PSNR and SSIM are incredibly low.

I'm quite sure this depends on the image size, as I trained/tested your network on images belonging to the original MIT-Adobe-5K-DPE and everything was ok, but when I add ".resize((256,256))" to the "load_image" function within util.py I have the problem I've just mentioned. 

Any idea why this happens?

Thank you!
Claudio

**Comment by sjmoran** [comment]
Hi Claudio, firstly thank you for your interest in CURL. It sounds to me like a classic case of overfitting. 256x256 pixels is a reduced amount of data to what we originally trained the network on I believe. As a first step, I would recommend increasing  the regularisation strength in the Adam optimiser (line 197 in main.py, weight_decay) and also monitoring the validation loss during training. The training and validation loss should not diverge. You could also reduce the weights in the Unet but that would be a bit more work, so my suggestion would be to start with the weight decay parameter. Please post back here on how you get on and if it doesn’t work we can continue the investigation.

**Comment by claudiom4sir** [comment]
Hi,

I solved the problem. In data.py there is this instruction `if input_img.shape[1]==output_img.shape[2]: output_img=output_img.permute(0,2,1)` that, when using square images, always changes the rows with the columns of the output image. Therefore, I compared the network output with the ground truth rotated by 90 degrees, and that was why I obtained such poor results during the validation and test phases. After removing that instruction everything works as expected.

Thank you
Claudio
